src/binding.cpp

```diff
@@ -172,35 +171,32 @@ class JSDawg : public Nan::ObjectWrap
 
 struct dawg_search_result
 {
-    int node_offset;
-    bool found;
-    bool final;
-    int skipped;
-    int child_count;
-    char* match_string;
+    std::string match_string = "";
```

Comment:
I *think* my motivation here for making it a `char*` originally instead of a string was to save allocations because some of the search function variants don't use this field. I imagine that's probably not worth worrying about though?

Reply:
My assumption is that `std::string` will only allocate once it starts growing and even then it will use [SSO](https://github.com/mapbox/cpp/blob/master/glossary.md#small-size-optimization) to avoid allocations for small strings.

But your comment helped me realize that even if I am assuming right above, this change still made the size of the `dawg_search_result` go from `24` bytes to `32`. From the look of the code that does not seem to matter (we don't store hundreds of `dawg_search_result` in memory), but nevertheless I made 7f01512 to re-shrink the size of the struct - this works more like before without the need for manual allocation and delete. Feeling good about this approach with `std::unique_ptr`. Look okay?